arelle/plugin/validate/EDINET/rules/edinet.py

```diff
@@ -84,24 +86,86 @@ def rule_EC8033W(
+        if latestPriorYearContext is None or \
+                priorYearContext.endDatetime > latestPriorYearContext.endDatetime:
+            latestPriorYearContext = priorYearContext
+    if latestPriorYearContext is None:
+        return
     currentYearContexts = [
         context
         for contextId, context in val.modelXbrl.contexts.items()
         if contextId.startswith('CurrentYear')
            and context.startDatetime is not None
            and context.isStartEndPeriod
     ]
-    for priorYearContext, currentYearContext in itertools.product(priorYearContexts, currentYearContexts):
-        if priorYearContext.endDatetime > currentYearContext.startDatetime:
+    earliestCurrentYearContext = None
+    for currentYearContext in currentYearContexts:
+        if earliestCurrentYearContext is None or \
+                currentYearContext.endDatetime > earliestCurrentYearContext.startDatetime:
+            earliestCurrentYearContext = currentYearContext
+    if earliestCurrentYearContext is None:
+        return
+    if latestPriorYearContext.endDatetime > earliestCurrentYearContext.startDatetime:
+        yield Validation.warning(
+            codes='EDINET.EC8033W',
+            msg=_("The startDate element of the current year context (id=%(currentYearContextId)s) is "
+                  "set to a date that is earlier than the endDate element of the prior year context "
+                  "(id=%(priorYearContextId)s). Please check the corresponding context ID "
+                  "%(currentYearContextId)s and %(priorYearContextId)s. Set the startDate element of "
+                  "context ID %(currentYearContextId)s to a date that is later than or equal to the "
+                  "endDate element of context ID %(priorYearContextId)s."),
+            currentYearContextId=earliestCurrentYearContext.id,
+            priorYearContextId=latestPriorYearContext.id,
+            modelObject=priorYearContexts + currentYearContexts,
+        )
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[DISCLOSURE_SYSTEM_EDINET],
+)
+def rule_EC8062W(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    EDINET.EC8062W: The sum of all liabilities and equity must equal the sum of all assets.
+    """
+    deduplicatedFacts = pluginData.getDeduplicatedFacts(val.modelXbrl)
+    contextIdPattern = regex.compile(r'^(Prior[1-9]Year|CurrentYear|Prior[1-9]Interim|Interim)(Duration|Instant)$')
+
+    factsByContextId = defaultdict(list)
+    for fact in deduplicatedFacts:
+        if fact.concept.qname not in (pluginData.assetsIfrsQn, pluginData.liabilitiesAndEquityIfrsQn):
+            continue
+        if not contextIdPattern.match(fact.contextID):
+            continue
+        factsByContextId[fact.contextID].append(fact)
+
+    for contextId, facts in factsByContextId.items():
```

Comment:
nit: keys aren't used:
```suggestion
    for facts in factsByContextId.values():
```